### DISCUSSION TITLE
Add a way to specify a `term://` URI that opens the user's default shell (like `:terminal`)

### DISCUSSION THREAD

**Discussion by LunarLambda** [neovim/neovim#30147] [Ideas]
I would like to be able to run `nvim -O term:// file` to open a vertical split with a terminal buffer and a file.

Editing `term://` however exits instantly instead of using my shell. Typing `term://zsh` is quite lengthy to type and I don't want to actually hardcode a shell.

I'm not sure if `term://` opening a terminal buffer without any command (immediately resulting in `process exited with 0`) is useful, so maybe it should just default to matching the behaviour of `:terminal`.

**Comment by justinmk** [comment]
We need to define a better spec for the `term://` URI first ... https://github.com/neovim/neovim/issues/3278

**Reply by LunarLambda** [reply]
Alternatively, maybe a solution could be to support CLI args of the form `:cmd` which in combination with `-o`/`-O`/`-p` runs `:cmd` in the respective window/tabpage. This would allow opening things like `:help` pages, `:Man`, `:terminal`, `:InspectTree` while working on tree-sitter plugins, etc...

don't think `:path` means anything special on Windows, and it's quite unlikely that regular files would start with that.

**Reply by LunarLambda** [reply]
Actually, hm, I suppose it might be possible to implement that with just an autocmd.

**Reply by LunarLambda** [reply]
```lua
local augroup = vim.api.nvim_create_augroup
local autocmd = vim.api.nvim_create_autocmd

local augid = nil

augid = augroup('cli-cmd', { clear = true })

autocmd('BufReadCmd', {
    group = augid,
    pattern = ':*',
    callback = function(event)
        if vim.v.vim_did_enter == 1 then return end

        vim.cmd(event.file:sub(2))
        -- For some reason deleting the buffer causes the splits to get broken.
        -- But only during startup?
        vim.cmd.bdelete(event.buf)term
        return vim.v.vim_did_enter == 1
    end,
    desc = 'Open commands in windows on startup'
})
```

I was hoping it would be this simple, but unfortunately calling `bdelete` messes up the splits for some reason? I end up with a single window and two buffers. Weirdly if I do the same steps manually it works fine. Unsure if it's a bug or not.

**Reply by LunarLambda** [reply]
This version works, but I'm not sure why it works and the other one doesn't:

```lua
local augroup = vim.api.nvim_create_augroup
local autocmd = vim.api.nvim_create_autocmd

local augid = augroup('cli-cmd', { clear = true })

local bufs = {}

local id = autocmd('BufReadCmd', {
        group = augid,
        pattern = ':*',
        callback = function(event)
            -- if vim.v.vim_did_enter == 1 then return end
            bufs[event.buf] = { win = vim.api.nvim_get_current_win(), cmd = event.file:sub(2) }
            -- vim.api.nvim_win_call(vim.api.nvim_get_current_win(), function()
            --     vim.cmd(event.file:sub(2))
            --     vim.cmd.bdelete(event.buf)
            -- end)
            --
            -- return vim.v.vim_did_enter == 1
        end,
        nested = true,
        desc = 'Open Ex commands in splits/tabs on startup'
})

autocmd('VimEnter', {
    group = augid,
    callback = function(event)
        for buf, data in pairs(bufs) do
            vim.api.nvim_win_call(data.win, function()
                vim.cmd(data.cmd)
                vim.cmd.bdelete(buf)
            end)
        end

        vim.api.nvim_del_autocmd(id)
    end,
    once = true,
    nested = true,
    desc = 'Open Ex commands in splits/tabs on startup'
})
```

This also has issues however, such as autocommands firing in the wrong buffer/window (fixable probably)
